### ISSUE TITLE
Wrong Percentage Check

### ISSUE THREAD

**Issue by StephanPraetsch** [logzio/logzio-logback-appender#12]
Hi,

you have a bug in your LogzioSender: If my tmp-dir has more than 98% percent free it will drop immediately the logs:

11:42:31,493 |-WARN in io.logz.logback.LogzioLogbackAppender[LOGZIO] - Logz.io: Dropping logs, as FS free usable space on /tmp/custom.checks.server/logzio-buf is 99 percent, and the drop threshold is 98 percent

The comparison operator is wrong at https://github.com/logzio/logzio-logback-appender/blob/8dd5f2e8a0a38a347acb57e3882bc2a0e0fa5389/src/main/java/io/logz/logback/LogzioSender.java#L198

Instead of 

if (actualFsPercent >= fsPercentThreshold) {

it must be

if (actualFsPercent < fsPercentThreshold) {

Kind regards
Stephan

**Comment by roiravhon** [comment]
Hey @StephanPraetsch,
This is the intended behaviour...

Our goal is not to flood the FS in case there are too many logs and there is no space left.
So we are calculating the current FS size into "actualFsPercent", and then checking that larger or equal to the configured "fsPercentThreshold".

If we did hit the checkmark, we are dropping the log.

Please let me know if that makes sense to you, and if you think something in wrong in the documentation.

**Comment by MaxKuehn** [comment]
Hiho,

> The percent of **used file system space** at which the appender will stop buffering. When we will reach that percentage, the file system in which the buffer rests will drop all new logs **until the percentage of used space drops below that threshold**. Set to -1 to never stop processing new logs

The way I read this is: If the directory fuller (i.e. >=) than the given threshold, throw everything new away. Keep throwing it away until it sinks below the threshold again.

But the function you use in the calculation of the actualFsPercent variable gives you the number of bytes still available. 
cf. [file_getusablespace.htm](http://www.tutorialspoint.com/java/io/file_getusablespace.htm)
That means that actualFsPercent holds the percentage of free space and the code does the following.
> Is more than xx% space **free**? Yes! Do we have enough? No! (<- the WTF part) Throw everything new away.
when it should be
> Is more than xx% space **used**? Yes! Do we have enough? No! Throw everything new away.

Assuming the documentation the way I read it is the intended behaviour, I'd either calculate
```
 actualFsPercent = 1.0 - ((queueDirectory.getUsableSpace() / queueDirectory.getTotalSpace()) * 100)
```
(and maybe rename it to actualUsedFsPercent or something)
(I didn't find a getUsedSpace()...)
**OR**
Calculate a new fileSystemNotFullPercentThreshold
```
fileSystemNotFullPercentThreshold = 100 - fileSystemFullPercentThreshold
```
and Change the comparison operator to "<" so you get
> Have we have less than (let's say) 2% left? Yes! Do we have enough left? No!
(and maybe rename actualFsPercent to actualFreeFsPercent)

I like the first one better. Fewer changes and you pretty much code what the doc says.

P.S.: The corresponding test might be wrong. I can't run any test, because I'm using Windows and the test enviroment uses some posix file attributes stuff. But that's basically my fault. The error is "using Windows".
P.P.S.:
(*pants down*) I'm working with @StephanPraetsch, that's why I'm barging in. He literally had to create bogus files to get his /tmp used percentage up to 98% on a newly set up workstation. The appender wouldn't log otherwise. (Yes we know, set the threshold to -1, but we wanted to know "why"!)

**Comment by roiravhon** [comment]
Thanks @MaxKuehn and @StephanPraetsch!
Fixed the bug, and fixed the test so it will point out an issue like that in the future.

PR pending.

**Comment by roiravhon** [comment]
Published the fix to maven central (1.0.8), usually takes 1-2 hours to propagate.
Thanks!
